pkg/manifests/config_test.go

```diff
@@ -25,54 +25,220 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func TestConfigParsing(t *testing.T) {
-	f, err := os.Open("testdata/cluster-monitoring-configmap.yaml")
-	if err != nil {
-		t.Fatal(err)
-	}
-	c, err := NewConfig(f, false)
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	if c.ClusterMonitoringConfiguration.AlertmanagerMainConfig.VolumeClaimTemplate == nil {
-		t.Fatal("config parsing failed: AlertmanagerMainConfig VolumeClaimTemplate was not parsed correctly")
-	}
-}
-
-func TestNewUserConfigFromStringParsing(t *testing.T) {
-	c, err := os.ReadFile("testdata/user-workload-configmap.yaml")
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	uwmc, err := NewUserConfigFromString(string(c))
-	if err != nil {
-		t.Fatal(err)
+func TestNewConfigFromString(t *testing.T) {
```

Comment:
side-note: I'm a fan of gotest.tools/v3/golden for this kind of test.

Reply:
thanks. yep, very useful when storing the expected, because the strings are input and it's easier when input/expected is easily reachable (no need to look for the file and go back and forth between the test and the file) I tend to store everything in one place, especially when the strings are small and won’t overload the test.